File: src/main/java/com/wanderers/hotelier_webservices/server/exception/CustomerServiceException.java
package com.wanderers.hotelier_webservices.server.exception;

import com.wanderers.hotelier_webservices.server.service.api.CustomerService;

/**
 * Custom Exception class which used to throw failures in {@link CustomerService}
 */
public class CustomerServiceException extends ServerException {

    private static final long serialVersionUID = 1L;

    /**
     * Class constructor
     *
     * @param message String value of Exception
     */
    public CustomerServiceException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/server/exception/CustomerServiceException.kt
package com.wanderers.hotelier_webservices.server.exception

import com.wanderers.hotelier_webservices.server.service.api.CustomerService

/**
 * Custom Exception class which used to throw failures in [CustomerService]
 */
class CustomerServiceException
/**
 * Class constructor
 *
 * @param message String value of Exception
 */
    (message: String?, cause: Throwable?) :
    ServerException(message, cause) {
    companion object {
        private const val serialVersionUID = 1L
    }
}